File: base/src/main/java/messyellie/storagecores/base/data/tiers/ItemProperties.java
package messyellie.storagecores.base.data.tiers;

import com.mojang.serialization.Codec;
import com.mojang.serialization.codecs.RecordCodecBuilder;

// todo: determine final set of properties
public record ItemProperties(boolean fireResistant) {
    public static final Codec<ItemProperties> CODEC = RecordCodecBuilder.create(
            instance -> instance.group(
                            Codec.BOOL.optionalFieldOf("fire_resistant", Boolean.FALSE).forGetter(ItemProperties::fireResistant)
                    )
                    .apply(instance, ItemProperties::new)
    );
}

File: base/src/main/kotlin/compass_system/storagecores/base/data/tiers/ItemProperties.kt
package compass_system.storagecores.base.data.tiers

import com.mojang.serialization.Codec
import com.mojang.serialization.codecs.RecordCodecBuilder

data class ItemProperties(val fireResistant: Boolean) {
    companion object {
        val codec: Codec<ItemProperties> = RecordCodecBuilder.create { instance ->
            instance.group(
                Codec.BOOL.optionalFieldOf("fire_resistant", false).forGetter(ItemProperties::fireResistant)
            ).apply(instance, ::ItemProperties)
        }
    }
}
